sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -172,7 +177,7 @@ def __init__(self, feature_id, enabled, filters=[], **kwargs):  # pylint: disabl
 
     def _validate(self):
         # type: () -> None
-        if not self.key.startswith(self.key_prefix):
+        if not self.feature_id.startswith(self.key_prefix):
```

Comment:
feature_id does not contain key_prefix.

Reply:
For now it does. I will make the key and feature_id swap in a future (probably separate PR). Right now feature_id is what key should be.